lib/bosh/BoshDirectorClient.js

```diff
@@ -45,19 +151,28 @@ class BoshDirectorClient extends HttpClient {
 
   /* Deployment operations */
 
-  getDeployments() {
-    return this
-      .request({
-        method: 'GET',
-        url: '/deployments'
-      }, 200)
-      .then(res => JSON.parse(res.body));
+  getDeployments(queryInactive) {
```

Comment:
refactor this method like this. 
`getDeployments()` -- returns all deployments in Primary config
`getDeploymentByConfig(boshDirectorConfig)` -- returns deployments in the bosh config as input

`getDeployments` would loop through all primary director config and invoke `getDeploymentByConfig` for each.